src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -193,28 +338,55 @@ private boolean containsKey(final String k) {
      * FeatureInputs constructed some other way will not be recognized by the engine.
      */
     @VisibleForTesting
-    public FeatureInput(final String name, final Map<String, String> kevValueMap, final File featureFile) {
+    public FeatureInput(final String name, final Map<String, String> keyValueMap, final String featureFile) {
         Utils.nonNull(name, "name");
-        Utils.nonNull(kevValueMap, "kevValueMap");
+        Utils.nonNull(keyValueMap, "kevValueMap");
         Utils.nonNull(featureFile, "featureFile");
         this.name = name;
-        this.kevValueMap = Collections.unmodifiableMap(new LinkedHashMap<>(kevValueMap));   //make a unmodifiable copy
+        this.keyValueMap = Collections.unmodifiableMap(new LinkedHashMap<>(keyValueMap));   //make a unmodifiable copy
         this.featureFile = featureFile;
         this.featureType = null;  // Must be set after construction
     }
 
+    public FeatureInput(final String name, Map<String, String> keyValueMap, File featureFile){
+        this(name, keyValueMap, featureFile.getAbsolutePath());
+    }
+
+    public FeatureInput(final String name, final FeatureCodec<T,?> codec, final File featureFile){
```

Comment:
`FeatureInput` is not supposed to be publicly-instantiable like this, as per the class-level comment. If you must have a public constructor for testing purposes, mark it `@VisibleForTesting`